delta/rdf/src/main/scala/ch/epfl/bluebrain/nexus/delta/rdf/graph/Graph.scala

```diff
@@ -85,8 +97,54 @@ final case class Graph private (root: IRI, model: Model) { self =>
     */
   def add(triple: Set[Triple]): Graph = {
     val stmt = triple.foldLeft(Vector.empty[Statement]) { case (acc, (s, p, o)) => acc :+ createStatement(s, p, o) }
-    Graph(root, model.add(stmt.asJava))
+    Graph(root, copy(model).add(stmt.asJava))
   }
+
+  /**
+    * Attempts to convert the current Graph to the N-Triples format: https://www.w3.org/TR/n-triples/
+    * @return
+    */
+  def toNTriples: Either[GraphError, NTriples] =
+    Try {
+      RDFWriter.create().lang(Lang.NTRIPLES).source(model).asString()
+    }.toEither.map(NTriples(_, root)).leftMap(err => ConversionError(Lang.NTRIPLES.getName, err.getMessage))
+
+  /**
+    * Attempts to convert the current Graph to the DOT format: https://graphviz.org/doc/info/lang.html
+    * The output will be in expanded form.
+    */
+  def toDot: Either[GraphError, Dot] =
+    toDot(ExtendedJsonLdContext.empty)
+
+  /**
+    * Attempts to convert the current Graph with the passed ''context'' as Json to the DOT format: https://graphviz.org/doc/info/lang.html
+    * The context will be inspected to populate its fields and then the conversion will be performed.
+    */
+  def toDot(
+      context: Json
+  )(implicit api: JsonLdApi, resolution: RemoteContextResolution, opts: JsonLdOptions): IO[GraphError, Dot] =
+    api.context(context, ContextFields.Include).leftMap(JsonLdErrorWrapper).flatMap(ctx => IO.fromEither(toDot(ctx)))
+
+  /**
+    * Attempts to convert the current Graph with the passed ''context'' as Json to the DOT format: https://graphviz.org/doc/info/lang.html
+    * The context will be inspected to populate its fields and then the conversion will be performed.
+    */
+  def toDot(
+      context: RawJsonLdContext
+  )(implicit api: JsonLdApi, resolution: RemoteContextResolution, opts: JsonLdOptions): IO[GraphError, Dot] =
+    toDot(Json.obj(keywords.context -> context.value))
+
+  /**
+    * Attempts to convert the current Graph with the passed ''context'' as Json to the DOT format: https://graphviz.org/doc/info/lang.html
+    * If the context is empty the DOT output will have the expanded form
+    */
+  def toDot(context: ExtendedJsonLdContext): Either[GraphError, Dot]                                        =
+    Try {
+      RDFWriter.create().lang(DOT).source(model).set(ROOT_ID, rootResource).set(JSONLD_CONTEXT, context).asString()
```

Comment:
RDFWriter was also in your previous PR, maybe we could have an utility method which would take care about wrapping this in an Either ?